### ISSUE TITLE
hillshade shadow and highlight colors wont behave correctly

### ISSUE THREAD

**Issue by farfromrefug** [CartoDB/mobile-sdk#412]
PLaying with the hillshade tile layer i realized that it wasnt behaving correctly.
Using the same highlight (accent in Mapbox) color and shadow color you would expect to get the same render as with mapbox. However it behave strangely.

Especially changing the highlight/shadow colors alpha. 
As you can see in the video changing the alpha of either highlight or shadow color seems to "saturate" instead of rendering the feature (shadow or highlight) transparent.

https://user-images.githubusercontent.com/655344/103900882-a6792200-50f8-11eb-9ed1-82be8909d307.mp4

I am working on fixing this. The idea is get something as close to mapbox as possible. So i look at this https://github.com/mapbox/mapbox-gl-js/blob/main/src/shaders/hillshade.fragment.glsl.

I will also try to add the accent color. I need to help from you. First maybe you have an idea on how we could fix it?

Thanks

**Comment by mtehver** [comment]
You may be correct here, it seems that the alpha channel is not handled correctly here. You can try pre-multiplying with alpha. For example in TileRenderer.cpp, do something like this:

```
            vt::GLTileRenderer::LightingShader lightingShaderNormalMap(false, LIGHTING_SHADER_NORMALMAP, [this](GLuint shaderProgram, const vt::ViewState& viewState) {
                float shadowAlpha = _normalMapShadowColor.getA() / 255.0f;
                glUniform4f(glGetUniformLocation(shaderProgram, "u_shadowColor"), _normalMapShadowColor.getR() * shadowAlpha / 255.0f, _normalMapShadowColor.getG() * shadowAlpha / 255.0f, _normalMapShadowColor.getB() * shadowAlpha / 255.0f, shadowAlpha);
                float highlightAlpha = _normalMapHighlightColor.getA() / 255.0f;
                glUniform4f(glGetUniformLocation(shaderProgram, "u_highlightColor"), _normalMapHighlightColor.getR() * highlightAlpha / 255.0f, _normalMapHighlightColor.getG() * highlightAlpha / 255.0f, _normalMapHighlightColor.getB() * highlightAlpha / 255.0f, highlightAlpha);
                glUniform3fv(glGetUniformLocation(shaderProgram, "u_lightDir"), 1, _mainLightDir.data());
            });
```

**Comment by farfromrefug** [comment]
@mtehver it did fix it ! I ll create a PR

**Comment by mtehver** [comment]
Thanks for testing. No need for PR, I have already fixed it locally.

**Comment by mtehver** [comment]
Closing, this was fixed in SDK 4.3.4.
